### ISSUE TITLE
Creating DeepLabCut Environment in Red Hat Linux

### ISSUE THREAD

**Issue by nmtimme** [DeepLabCut/DeepLabCut#499]
**Your Operating system and DeepLabCut version**

Please state your operating system, env, and which version of DeepLabCut you are using.
Example: Ubuntu 16.04 LTS, with an Anaconda Env, & DeepLabCut1.x or 2.x.

OS: Red Hat Enterprise Linux (RHEL 7)
Anaconda Environment
DeepLabCut Version 2.1.1
Browser: Firefox

After running into problems on my windows machine  #463 , I've decided to try running deeplabcut on a cluster at my university. Unfortunately, this cluster uses Red Hat and not Ubuntu, so I haven't been using the provided Ubuntu environment. (I tried using it, but it didn't work.) I was advised by my university's IT staff to try building the environment myself. Unfortunately, that isn't working either, so I thought I would contact you to see if you can provide general directions on how to create the environment. 

For instance, if I try:

conda create -n DeepLabCutEnv python=3.6.8 tensorflow=1.13.1 imageio wxpython=4.0.3 matplotlib=3.0.3

The environment is created without errors or warnings. Inside the environment, I then run:

python setup.py install

(Note, I had to modify a line of the setup.py file. I changed:
'ipython','ipython-genutils','imageio~=2.3.0' ,
to:
'ipython','ipython-genutils','imageio~=2.6.1'
to avoid an error associated with imageio.)

I do get one error in that process (see below), but it runs to completion, so I think it found a workaround?

Installed .../envs/DeepLabCutEnvRN/lib/python3.6/site-packages/requests-2.22.0-py3.6.egg
Searching for pyyaml>=5.1
Reading https://pypi.org/simple/pyyaml/
Downloading https://files.pythonhosted.org/packages/e3/e8/b3212641ee2718d556df0f23f78de8303f068fe29cdaa7a91018849582fe/PyYAML-5.1.2.tar.gz#sha256=01adf0b6c6f61bd11af6e10ca52b7d4057dd0be0343eb9283c878cf3af56aee4
Best match: PyYAML 5.1.2
Processing PyYAML-5.1.2.tar.gz
Writing /tmp/1457920.s1/easy_install-4qugiixm/PyYAML-5.1.2/setup.cfg
Running PyYAML-5.1.2/setup.py -q bdist_egg --dist-dir /tmp/1457920.s1/easy_install-4qugiixm/PyYAML-5.1.2/egg-dist-tmp-xsejm5c5
In file included from ext/_yaml.c:596:0:
ext/_yaml.h:2:18: fatal error: yaml.h: No such file or directory
 #include <yaml.h>
                  ^
compilation terminated.
Error compiling module, falling back to pure Python
zip_safe flag not set; analyzing archive contents...
Moving PyYAML-5.1.2-py3.6-linux-x86_64.egg to .../envs/DeepLabCutEnvRN/lib/python3.6/site-packages
Adding PyYAML 5.1.2 to easy-install.pth file

Then, when I attempt to import deeplabcut inside ipython (still in the environment), I get the following error:

ImportError: Cannot load backend 'TkAgg' which requires the 'tk' interactive framework, as 'headless' is currently running

I've seen in previous issues that this can be fixed with matplotlib=3.0.3, but I already included that in the environment. Furthermore, if I try to go back and install matplotlib=3.0.3 with pip, I get the response that it is already installed. 

I understand that I'm working on a bit of a strange system, but I'd appreciate any advice you can provide. I'm new to python, so I appreciate all the help I can get. Thanks!

~Nick

**Comment by xdhmoore** [comment]
Hi Nick,

I am doing the same thing at my school. It's a work in progress but it looks like if you set
`export DLClight=True`
in bash, it will get past the TkAgg error, though of course you can't use the GUI in light-mode. Note that `l` in `light` is lowercase. I myself did not note this and there was great confusion.

Good luck.

**Comment by nmtimme** [comment]
Thanks! I found that when I executed "export DLClight=True" in the conda environment, I could import deeplabcut successfully. As you say, it now runs in light-mode, which I don't totally understand at this point. However, so long as I do most things off the cluster (e.g., label videos) and only do the training on the cluster, that should work. It would be nice if there were a better solution to this problem, though.
